icu4c/source/i18n/rbtz.cpp

```diff
@@ -114,32 +121,35 @@ RuleBasedTimeZone::operator!=(const TimeZone& that) const {
 
 void
 RuleBasedTimeZone::addTransitionRule(TimeZoneRule* rule, UErrorCode& status) {
+    LocalPointer<TimeZoneRule>lpRule(rule);
```

Comment:
good -- but we should document in the coding guidelines an exception to the “bail out immediately for incoming error” rule, saying that if a function adopts objects, it has to do so even if there is an incoming error.

separate PR or ticket?

Reply:
I think a separate PR, and if we have some generic user guide update ticket, do it using that.

I can do it.

Reply:
> I think a separate PR, and if we have some generic user guide update ticket, do it using that.

Separate PR is fine. If you can do it quickly, please use this same ticket.